nomad/structs/structs.go

```diff
@@ -2253,7 +2330,28 @@ func (n *NodeResources) Equals(o *NodeResources) bool {
 	return true
 }
 
-// DevicesEquals returns true if the two device arrays are equal
+func (n *Networks) Equals(o *Networks) bool {
+	if n == nil && o == nil {
+		return true
+	}
+	if n == nil || o == nil {
+		return false
+	}
+	if len(*n) != len(*o) {
+		return false
+	}
+	oa := *o
+	// REVIEW: should this be set equality?
+	// in current code len is always 1 or 0
```

Comment:
I think it's possible to have multiple networks but that's rare.

Reply:
I think we should do set equality to avoid leaving a trap for ourselves if multiple networks become more common. It would be too easy to simple testing of multiple networks to not catch the bug in this code by just happening to be in the proper order.